### PR TITLE
Single page feedback form

### PR DESCRIPTION
**Pull request by stevehook**
## Context

We want to replace the current multi-step survey form with a simpler one page form, that is based on the GDS feedback form.

This PR implements the new form behind a `feedback_form` feature flag.

## Changes proposed in this pull request

- [x] `feedback_form` feature flag
- [x] New routes, controller, form object for the new form that is swapped in depending on the feature flag.
- [x] System spec to cover the new flow.

image

Note that the migration that this work depends on has already deployed to production. There will be a further PR to add a data export for the new feedback form (details TBC). Finally we can remove the old satisfaction survey views, controller, forms etc. when the `feedback_form` feature flag is retired.

## Guidance to review

- It should be possible to manually test the new form by enabling the feature flag and submitting an application. You can also see the interface by navigating direct to `/candidate/application/feedback-form`
- Note that we don't prevent visits to the feedback form via the URL outside the normal application submission process.

## Link to Trello card

https://trello.com/c/lQdpvtmy/2281-dev-replace-multi-step-survey-with-single-page-feedback-form

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/candidate_interface/feedback_form/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<% content_for :title, title_with_error_prefix(t('page_titles.your_feedback'), @feedback_form.errors.any?) %>
+<% content_for :before_content, govuk_back_link_to(candidate_interface_application_submit_success_path) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: @feedback_form, url: candidate_interface_create_feedback_form_path, method: :post do |f| %>
+      <%= f.govuk_error_summary %>
+      <h2 class="govuk-heading-xl">
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/feedback_form/new.html.erb:8]:
```suggestion
      <h1 class="govuk-heading-xl">
```

File: app/views/candidate_interface/feedback_form/thank_you.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<% content_for :title, t('page_titles.thank_you') %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+
+    <h2 class="govuk-heading-xl">
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/feedback_form/thank_you.html.erb:6]:
```suggestion
    <h1 class="govuk-heading-xl">
```

File: config/locales/feedback_form.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+en:
+  satisfaction_levels:
+    very_satisfied: 'Very satisfied'
+    satisfied: 'Satisfied'
+    neither_satisfied_or_dissatisfied: 'Neither satisfied or dissatisfied'
+    dissatisfied: 'Dissatisfied'
+    very_dissatisfied: 'Very dissatisfied'
```
Comment (paulrobertlloyd) [inline config/locales/feedback_form.yml:7]: Sometimes I see YAML strings quoted, sometimes not. I know you need to do this if they include `:`, but that’s not the case here. Not sure what our code style is for YAML.

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]

File: config/locales/feedback_form.yml (hunk 3 above)
Comment (stevehook) [reply config/locales/feedback_form.yml]: I don't think we have a written style for this but most people only add quotes when needed. Less noise more readable I guess. So this is a mistake, I've removed them.

File: config/routes.rb [hunk 4]
```diff
@@ -400,6 +400,12 @@
         post '/complete' => 'safeguarding#complete', as: :complete_safeguarding
       end
 
+      scope '/feedback-form' do
+        get '/' => 'feedback_form#new', as: :feedback_form
+        post '/' => 'feedback_form#create', as: :create_feedback_form
```
Comment (malcolmbaig) [inline config/routes.rb:405]: Minor suggestion - not super-important as it's not something we're consistent with in the routes, but we don't need the `create_feedback_form_path`. Can remove it and just POST to the `feeback_form_path` in the form for the same result.

**Review by malcolmbaig — APPROVED** [review]
LG :+1:

File: app/views/candidate_interface/feedback_form/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<% content_for :title, title_with_error_prefix(t('page_titles.your_feedback'), @feedback_form.errors.any?) %>
+<% content_for :before_content, govuk_back_link_to(candidate_interface_application_submit_success_path) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: @feedback_form, url: candidate_interface_create_feedback_form_path, method: :post do |f| %>
+      <%= f.govuk_error_summary %>
+      <h1 class="govuk-heading-xl">
+        <%= t('page_titles.your_feedback') %>
+      </h2>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/feedback_form/new.html.erb:10]: Eek - my fault. Should have highlighted full heading tag.

```suggestion
      </h1>
```

File: app/views/candidate_interface/feedback_form/thank_you.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<% content_for :title, t('page_titles.thank_you') %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+
+    <h1 class="govuk-heading-xl">
+      <%= t('page_titles.thank_you') %>
+    </h2>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/feedback_form/thank_you.html.erb:8]: Ditto.

```suggestion
    </h1>
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]

**Review by paulrobertlloyd — APPROVED** [review]

**Pull request merged**